Lib/uuid.py

```diff
@@ -176,7 +176,7 @@ def __init__(self, hex=None, bytes=None, bytes_le=None, fields=None,
                             'or int arguments must be given')
         if hex is not None:
             hex = hex.replace('urn:', '').replace('uuid:', '')
-            hex = hex.strip('{}').replace('-', '')
+            hex = hex.strip("{}").replace("-", "").replace("_", "")
             if len(hex) != 32:
```

Comment:
Shouldn't we also check for a `0x` prefix like the json module?

https://github.com/python/cpython/blob/f203d1cb52f7697140337a73841c8412282e2ee0/Lib/json/decoder.py#L61

```suggestion
            if len(hex) != 32 or hex[1] in "xX":
```

Reply:
Good catch, also those are not valid UUID.

However I think that it would be better to lstrip those elements, to allow providing a UUID like `0x12345678123456781234567812345678` or `+12345678123456781234567812345678`, but now that I wrote it down it feels strange to me.

What do you thing @nineteendo?

Reply:
I would reject strings that don't match this regex: `r'[0-9A-Fa-f]{32}'`, supporting them is a feature that needs to be discussed.

Reply:
```suggestion
            if len(hex) != 32 or not re.match(r'[0-9A-Fa-f]{32}', hex)
```
I will use this check (and remove the underscore strip on the above line): the length check will easy get other errors and is faster that a regex, for everything else the regex matches any unwanted character before passing to `int`.

Reply:
You shouldn't worry about performance in case of an error:

```suggestion
            if not re.fullmatch(r'[0-9A-Fa-f]{32}', hex):
```

Reply:
Can you also add a test with unicode digits:

```python
>>> int("\uff10", 16)
0
```